projects/pre-sentence-reports-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/listener/MessageListener.kt

```diff
@@ -0,0 +1,33 @@
+package uk.gov.justice.digital.hmpps.listener
+
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.jms.annotation.EnableJms
+import org.springframework.jms.annotation.JmsListener
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.integrations.delius.document.DocumentService
+import uk.gov.justice.digital.hmpps.integrations.psr.PsrClient
+import uk.gov.justice.digital.hmpps.message.SimpleHmppsEvent
+import uk.gov.justice.digital.hmpps.telemetry.TelemetryService
+import java.net.URI
+
+@Component
+@EnableJms
+class MessageListener(
+    private val telemetryService: TelemetryService,
+    private val psrClient: PsrClient,
+    private val documentService: DocumentService
+) {
+
+    companion object {
+        val log: Logger = LoggerFactory.getLogger(this::class.java)
+    }
+
+    @JmsListener(destination = "\${spring.jms.template.default-destination}")
+    fun receive(hmppsEvent: SimpleHmppsEvent) {
+        log.info("received $hmppsEvent")
+        telemetryService.hmppsEventReceived(hmppsEvent)
+        val psrDocument = psrClient.getPsrReport(URI.create(hmppsEvent.detailUrl))
```

Comment:
Will detailUrl actually give us the PDF?  There may be other clients listening to these events, so I expected it to be a JSON endpoint and we'd have to construct the PDF URL ourselves.

Reply:
There's this one, but might need to confirm how accurate it is: https://github.com/ministryofjustice/hmpps-probation-integration-lambda-functions/blob/main/pre-sentence-report-handler/test_event.json